apps/docs/pages/guides/platform/database-usage.mdx

```diff
@@ -23,20 +23,44 @@ This value is also reported in the [database settings page](https://app.supabase
-For Pro and Enterprise projects, disk size expands ~1.5x automatically (e.g., 8GB -> 12GB) when you reach 90% of the disk size.
-Disk size expansion can only occur once every six hours.
-Pro projects can store up to 1024TB.
 
-All projects enter read-only mode when you reach 95% of the disk size. In read-only mode, clients will encounter errors such as `cannot execute INSERT in a read-only transaction`.
+For Pro and Enterprise tier projects, [disk size expands ~1.5x automatically](#paid-tier-automatic-disk-resizing) when you reach 90% of the disk size. Disk size expansion can only occur once every six hours.
+
+For Free tier projects, the database will enter ['read only' mode when you reach 95% disk space](#free-tier-project-read-only-mode).
+
+Pro and enterprise tier projects can store up to 1024TB.
+
+### Free tier project 'read-only' mode
+
+Free tier projects enter read-only mode when you reach 95% of the disk size.
+In read-only mode, clients will encounter errors such as `cannot execute INSERT in a read-only transaction`.
 Regular operation (read-write mode) is automatically re-enabled once usage is below 95% of the disk size.
 
+For [automatic disk resizing](paid-tier-automatic-disk-resizing), and avoid your database entering read-only mode, you can [upgrade to the Pro or Enterprise plan](https://app.supabase.com/project/_/settings/billing/subscription).
+
+### Paid tier automatic disk resizing
+
+If you are on Pro or Enterprise, database storage expands automatically when you reach 90% of the disk size. Usually the disk is expanded to be 50% larger (e.g., 8GB -> 12GB).
+For every automatic database storage expansion, it will not auto expand for another 6 hours.
+
+If you intend to import a lot of data into your database that would require multiple storage expansions then [reach out to our team](https://app.supabase.com/support/new), otherwise your project might enter a ['read-only' mode](#paid-tier-project-read-only-mode).
+
 If you need more than 1024TB of disk size or require multiple storage expansions in a short period of time, [contact us](https://app.supabase.com/support/new) to learn more about the Enterprise plan.
 
+#### Paid tier project 'read-only' mode
+
+During the 6 hour period after an automatic resize, your database will not automatically resize until after 6 hours.
+If within those 6 hours you reach 95% of the disk space, your project will enter read-only mode.
+
+In read-only mode, clients will encounter errors such as `cannot execute INSERT in a read-only transaction`.
+Regular operation (read-write mode) is automatically re-enabled once usage is below 95% of the disk size, or when the database automatically resizes after the 6 hour period.
+
 ### Increasing available disk size
 
 1. [Upgrade to the Pro or Enterprise plan](https://app.supabase.com/project/_/settings/billing/subscription) to increase your quota and expand your disk size automatically.
```

Comment:
Can we clarify that this is an either or? either upgrade to pro to not worry about this or reduce your disk usage

Reply:
great point.

Reply:
i'll change this in a commit.

Reply:
need a rereview 😄